spec/server/update_model_spec.rb

```diff
@@ -0,0 +1,87 @@
+describe UpdateModelController do
+  include Rack::Test::Methods
+
+  def app
+    OUTER_APP
+  end
+
+  let(:proper_name) { 'name' }
+  let(:improper_name) { 'namex' }
+
+  it "rejects requests from non-superusers" do
+    auth_header(:editor)
+
+    json_post(:update_model, {
+      project_name: "labors",
+      actions: [
+        action_name: "update_attribute",
+        model_name: "monster",
+        attribute_name: "name",
+        description: "The monster's name",
+      ]
+    })
+
+    expect(last_response.status).to eq(403)
+  end
+
+  it "updates attribute options" do
+    auth_header(:superuser)
+    json_post(:update_model, {
+      project_name: "labors",
+      actions: [{
+        action_name: "update_attribute",
+        model_name: "monster",
+        attribute_name: proper_name,
+        description: "The monster's name",
+        display_name: "NAME"
+      }]
+    })
+
+    expect(last_response.status).to eq(200)
+    expect(Labors::Monster.attributes[:name].description).to eq("The monster's name")
+    expect(Labors::Monster.attributes[:name].display_name).to eq("NAME")
+
+    response_json = JSON.parse(last_response.body)
```

Comment:
nit: there is a `json_body()` method in `spec_helpers` that does this parsing of `last_response` for you, so that could save some writing in these specs.